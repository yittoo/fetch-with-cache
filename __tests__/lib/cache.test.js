"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const cache_1 = require("../../src/lib/cache");
describe('Cache Class', () => {
    let cache;
    beforeEach(() => {
        cache = new cache_1.Cache();
    });
    describe('constructor', () => {
        it('initialization store matches previous', () => {
            const storedData = cache.getStore();
            expect(storedData).toMatchSnapshot();
        });
    });
    describe('cleanAllCache', () => {
        it('cleans all cache', () => {
            cache.saveToCache('GET', 'url', 'data');
            expect(cache.getStore()).toMatchSnapshot();
            cache.cleanAllCache();
            expect(cache.getStore()).toMatchSnapshot();
        });
    });
    describe('saveToCache', () => {
        it('saves to cache', () => {
            cache.saveToCache('POST', 'url', 'data');
            expect(cache.getStore()).toMatchSnapshot();
        });
    });
    describe('saveToCache', () => {
        it('saves to cache', () => {
            cache.saveToCache('POST', 'url', 'data');
            expect(cache.getStore()).toMatchSnapshot();
        });
    });
    describe('checkCache', () => {
        it('return true if exists in cache', () => {
            cache.saveToCache('POST', 'url-existant', 'data');
            const exists = cache.checkCache('POST', 'url-existant');
            expect(exists).toBe(true);
        });
        it('return false if does exists in cache', () => {
            const exists = cache.checkCache('POST', 'url-inexistant');
            expect(exists).toBe(false);
        });
    });
});
