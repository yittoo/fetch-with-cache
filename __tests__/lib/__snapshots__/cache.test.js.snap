// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cache Class cleanAllCache cleans all cache 1`] = `
Object {
  "cache": Object {
    "CONNECT": Object {},
    "DELETE": Object {},
    "GET": Object {
      "url": "data",
    },
    "HEAD": Object {},
    "OPTIONS": Object {},
    "POST": Object {},
    "PUT": Object {},
    "TRACE": Object {},
    "TRACK": Object {},
  },
}
`;

exports[`Cache Class cleanAllCache cleans all cache 2`] = `
Object {
  "cache": Object {
    "CONNECT": Object {},
    "DELETE": Object {},
    "GET": Object {},
    "HEAD": Object {},
    "OPTIONS": Object {},
    "POST": Object {},
    "PUT": Object {},
    "TRACE": Object {},
    "TRACK": Object {},
  },
}
`;

exports[`Cache Class constructor initialization store matches previous 1`] = `
Object {
  "cache": Object {
    "CONNECT": Object {},
    "DELETE": Object {},
    "GET": Object {},
    "HEAD": Object {},
    "OPTIONS": Object {},
    "POST": Object {},
    "PUT": Object {},
    "TRACE": Object {},
    "TRACK": Object {},
  },
}
`;

exports[`Cache Class saveToCache saves to cache 1`] = `
Object {
  "cache": Object {
    "CONNECT": Object {},
    "DELETE": Object {},
    "GET": Object {},
    "HEAD": Object {},
    "OPTIONS": Object {},
    "POST": Object {
      "url": "data",
    },
    "PUT": Object {},
    "TRACE": Object {},
    "TRACK": Object {},
  },
}
`;

exports[`Cache Class saveToCache saves to cache 2`] = `
Object {
  "cache": Object {
    "CONNECT": Object {},
    "DELETE": Object {},
    "GET": Object {},
    "HEAD": Object {},
    "OPTIONS": Object {},
    "POST": Object {
      "url": "data",
    },
    "PUT": Object {},
    "TRACE": Object {},
    "TRACK": Object {},
  },
}
`;
